### PR TITLE
Split SelectEntityInteractor and SelectMessageInteractor

### PR DESCRIPTION
**Pull request by teamagam5**
Split SelectEntityInteractor and SelectMessageInteractor,
changed their usages accordingly.

### CODE REVIEW

File: domain/src/main/java/com/teamagam/gimelgimel/domain/messages/SelectMessageInteractor.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    private DataSubscriptionRequest buildSelectMessageRequest(
+            DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+        return factory.create(
+                Observable.just(mMessageId),
+                entityIdObservable ->
+                        entityIdObservable
+                                .flatMap(mMessagesRepository::getMessage)
+                                .filter(m -> m != null)
```
Comment (BarAgam) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/messages/SelectMessageInteractor.java]: I think it'll be best to log the cases where the interactor got executed with an ID that has no related message.

File: domain/src/main/java/com/teamagam/gimelgimel/domain/messages/repository/EntityMessageMapper.java [hunk 2]
```diff
@@ -8,11 +8,11 @@
 public interface EntityMessageMapper {
     void addMapping(String messageId, String entityId);
 
-    Observable<String> getEntityId(Message message);
+    String getEntityId(Message message);
```
Comment (BarAgam) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/messages/repository/EntityMessageMapper.java:11]: Good job

File: domain/src/main/java/com/teamagam/gimelgimel/domain/map/SelectEntityInteractor.java [hunk 3]
```diff
@@ -52,23 +44,7 @@ protected SelectEntityInteractor(
     @Override
     protected Iterable<SubscriptionRequest> buildSubscriptionRequests(
```
Comment (BarAgam) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/map/SelectEntityInteractor.java:50]: In buildSelectEntityRequest it's assumed that the given ID will be successfully mapped to an entity.
I think it'll be best to check for it and log the cases where there's no entity mapped to the given ID.
[It's something I wrote if I'm not mistaking... my bad...]

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/MapViewModel.java [hunk 4]
```diff
@@ -155,6 +163,8 @@ public void display(VectorLayerPresentation vectorLayerPresentation) {
         @Override
         public void entityClicked(String entityId) {
             mSelectEntityInteractorFactory.create(entityId).execute();
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/MapViewModel.java:165]: We could've made an entirely new interactor here that does those two interactor calls.
I leave it up to you to decide if you want to do it now or not.

**Review by BarAgam — APPROVED** [review]

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/MapViewModel.java (hunk 4 above)
Comment (teamagam5) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/MapViewModel.java:165]: Yeah I know. I decided not to...

**Pull request merged**